### ISSUE TITLE
Memory not cleared in check_labels

### ISSUE THREAD

**Issue by backyardbiomech** [DeepLabCut/DeepLabCut#652]
**Describe the bug**
When checking labels on a large number of training images, memory does not appear to be cleared, and computer may crash depending on available resources.

**To Reproduce**
1. Create a project with a large number of labeled training images
2. run deeplabcut.check_labels(config)
3. Watch resource use (activity monitor on Mac). On my laptop, python hit 6 GB usage before the computer crashed. 

**Expected behavior**
After each video folder, or each plotted image is saved, the memory should be cleared. 

**Desktop (please complete the following information about your system):**
 - OS: macOS 10.15.4
 - DeepLabCut Version 2.1.6.4 (python 3.7.6)

**Additional context**
I tried editing MakeLabeledPlots() by adding:

```
import gc
...
plt.cla()
plt.clf()
plt.close("all")
gc.collect()
```

which did not fix the issue.

**Comment by AlexEMG** [comment]
Good point, I suppose we should just close the images that are creaed in the loop.... https://github.com/AlexEMG/DeepLabCut/blob/master/deeplabcut/utils/visualization.py#L71

**Comment by backyardbiomech** [comment]
That's in there

https://github.com/AlexEMG/DeepLabCut/blob/2e9003ce1304eccb5776a51b432ac047e012aee8/deeplabcut/generate_training_dataset/trainingsetmanipulation.py#L353

but doesn't seem to do the trick, at least on my computer.

**Comment by jeylau** [comment]
@haliaetus13, if that's a pressing issue, please have a look at `make_labeled_images_from_dataframe` in the branch 'fast_image_writing'.
https://github.com/AlexEMG/DeepLabCut/blob/5dc2cc90df97e95d34f7a604f5349ef28c06be59/deeplabcut/utils/visualization.py#L102
Labeled frames are written to disk about 3–4 times faster, and with constant memory usage (Activity Monitor indicated about 150 mb when I tried it with 2000 images).
This is still a functionality under development though, and will be incorporated later in the main package.

**Comment by backyardbiomech** [comment]
Thanks for pointing me to that @jeylau. I first tried a minor edit just converting the existing check_labels to the named-object based plotting, and that did not help. So then I converted the rest of the code to reuse the fig and ax objects based on the code you referenced, and that seemed to fix it. I'm happy to submit a PR if you want, unless the new release is coming soon enough.

There was a third, more literally copied, version I tried, when I tried loading the images into an image collection. That failed with a "read of closed file exception", which I replicated outside of deeplabcut, and with multiple image paths (with and without spaces, etc). Just something to be aware of before releasing.

I'm really looking forward to (i.e. NEED) that new multi-animal ability!
